ledger/src/blockstore_processor.rs

```diff
@@ -308,6 +311,7 @@ pub fn process_blockstore_from_root(
     bank.set_entered_epoch_callback(solana_genesis_programs::get_entered_epoch_callback(
         genesis_config.operating_mode,
     ));
+    bank.set_hard_forks(opts.hard_forks.clone());
```

Comment:
I think this `bank` from snapshots may contain a non-empty `hard_forks` because `Bank::hard_forks` is not `#[serde(skip)]`. In such a case, a warning or error is preferred when any discrepancy is detected unless `HardForksMap::default()` or subset?

Reply:
sgtm, we now emit a `warn!()` when the bank hard forks list changes